src/proxy.js

```diff
@@ -102,66 +103,88 @@ module.exports = (proxyTarget, host) => {
+						} else if (name === 'style') {
+							value = value.replace(cssUrlPattern, ($0, $1, $2, $3) => {
+								return `url("${rewrite($2 || $3, req.baseUrl)}")`;
+							});
+						} else {
+							value = rewrite(value, req.baseUrl);
+						}
+
+						el.setAttribute(name, value);
+					} catch (e) {}
+				});
+			},
+		};
+	})).concat(rewriteElements.map((name) => {
+		return {
+			query: name,
+			func (el, req) {
+				let value = '';
+				let stream = el.createStream()
+					.on('error', () => stream.end())
+					.on('data', chunk => value += chunk.toString())
+					.on('end', () => {
+						try {
+							stream.end(rewrite(value, req.baseUrl));
+						} catch (e) {
+							stream.end(value);
+						}
+					});
+			},
+		};
+	})).concat(rewriteRegExp.map((name) => {
+		return {
+			query: name,
+			func (el, req) {
+				let value = '';
+				let stream = el.createStream()
+					.on('error', () => stream.end())
+					.on('data', chunk => value += chunk.toString())
+					.on('end', () => {
+						try {
+							stream.end(rewriteAllAbsolute(value, req.baseUrl));
+						} catch (e) {
+							stream.end(value);
+						}
+					});
+			},
+		};
+	})), false)
+
 	return [
 		/**
-		 * Rewrite links in HTML.
+		 * Harmon middleware should only be applied to html and xml files.
 		 */
-		harmon([], rewriteAttributes.map((name) => {
-			return {
-				query: `[${name}]`,
-				func (el, req) {
-					el.getAttribute(name, (value) => {
-						try {
-							if (name === 'srcset') {
-								value = srcset.stringify(srcset.parse(value).map(src => (src.url = rewrite(src.url, req.baseUrl), src)));
-							} else if (name === 'style') {
-								value = value.replace(cssUrlPattern, ($0, $1, $2, $3) => {
-									return `url("${rewrite($2 || $3, req.baseUrl)}")`;
-								});
-							} else {
-								value = rewrite(value, req.baseUrl);
-							}
-
-							el.setAttribute(name, value);
-						} catch (e) {}
-					});
-				},
-			};
-		}).concat(rewriteElements.map((name) => {
-			return {
-				query: name,
-				func (el, req) {
-					let value = '';
-					let stream = el.createStream()
-						.on('error', () => stream.end())
-						.on('data', chunk => value += chunk.toString())
-						.on('end', () => {
-							try {
-								stream.end(rewrite(value, req.baseUrl));
-							} catch (e) {
-								stream.end(value);
-							}
-						});
-				},
-			};
-		})).concat(rewriteRegExp.map((name) => {
-			return {
-				query: name,
-				func (el, req) {
-					let value = '';
-					let stream = el.createStream()
-						.on('error', () => stream.end())
-						.on('data', chunk => value += chunk.toString())
-						.on('end', () => {
-							try {
-								stream.end(rewriteAllAbsolute(value, req.baseUrl));
-							} catch (e) {
-								stream.end(value);
-							}
-						});
-				},
-			};
-		})), true),
+		(req, res, next) => {
+			let path = req.path.toLowerCase();
+			let shouldParse = path.endsWith('/') || path.endsWith('.html') || path.endsWith('.xml');
```

Comment:
Looks ok at first glance but would still prefer matching based on the content-type. You can use https://github.com/jshttp/type-is#typeisismediatype-types which is already internally used by express as well.

Reply:
Here I am dealing with the `req`, but I can't find a way to get `req` headers including content-type, they are empty at that step.

Reply:
I am able to get type inside `func` callbacks. But that will do a lot of useless parsing, plus `return;` should be replaced with something smarter to return the content without any changes. 
```js
rewriteAttributes.map((name) => {
		return {
			query: `[${name}]`,
			func (el, req, res) {
				const shouldParse = typeis.is(res._headers['content-type'], ['text/xml', 'text/html']);
				if (!shouldParse) {
					return;
				}

				let value = '';
				let stream = el.createStream()
					.on('error', () => stream.end())
					.on('data', chunk => value += chunk.toString())
					.on('end', () => {
						stream.end(value);
					});
			},
		};
	})
```